include/groonga/groonga.h

```diff
@@ -906,6 +906,61 @@ GRN_API grn_obj *
 grn_ctx_db(grn_ctx *ctx);
 GRN_API grn_obj *
 grn_ctx_get(grn_ctx *ctx, const char *name, int name_size);
+/**
+ * \brief Retrieve all tables from the database context.
+ *
+ * \note Ensure that `tables_buffer` is properly initialized using
+ *       \ref GRN_PTR_INIT with the \ref GRN_OBJ_VECTOR flag before invoking
+ *       this function.
+ *
+ * \attention After processing the tables, it is essential to release the
+ *            allocated resources by unlinking each table object and finalizing
+ *            the `tables_buffer` using \ref GRN_OBJ_FIN. Neglecting to do so
+ *            can lead to memory leaks.
+ *
+ * Here is an example of how to use \ref grn_ctx_get_all_tables to retrieve and
+ * process all tables:
+ *
+ * ```c
+ * grn_rc rc;
+ * grn_obj tables;
+ * int i;
+ * int n_tables;
+ *
+ * // Initialize the tables_buffer as a pointer vector.
+ * GRN_PTR_INIT(&tables, GRN_OBJ_VECTOR, GRN_ID_NIL);
+ *
+ * // Retrieve all tables in the context.
+ * rc = grn_ctx_get_all_tables(ctx, &tables);
+ * if (rc != GRN_SUCCESS) {
+ *     GRN_OBJ_FIN(ctx, &tables);
+ *     // Handle error appropriately.
+ *     return rc;
+ * }
+ *
+ * // Calculate the number of tables retrieved.
+ * n_tables = GRN_BULK_VSIZE(&tables) / sizeof(grn_obj *);
+ * // Iterate over each table.
+ * for (i = 0; i < n_tables; i++) {
+ *     grn_obj *table = GRN_PTR_VALUE_AT(&tables, i);
+ *     // Use table.
+ * }
+ *
+ * // Free resources by unlinking each table.
+ * for (i = 0; i < n_tables; i++) {
+ *     grn_obj *table = GRN_PTR_VALUE_AT(&tables, i);
+ *     grn_obj_unlink(ctx, table);
```

Comment:
普通は`grn_obj_unref()`を使っておいたほうがいいです。

```suggestion
 *     grn_obj_unref(ctx, table);
```

Reply:
fix: [ea907da](https://github.com/groonga/groonga/pull/2129/commits/ea907dacc84e9024505e54b2aa808591f819272e)